### PR TITLE
Added Javadoc comments and fixed some typos

### PR DESCRIPTION
**Pull request by Speecker**
### **Describe your change:**

- Added: Javadoc comments
- Changed: made some minor typo fixes
- Fixed: missing package
- Changed: converted author and link comment into Javadoc comment

### CODE REVIEW

**Comment by Speecker** [comment]
resolved conflicts

**Comment by Speecker** [comment]
resovled build errors but FIX TREETRAVERSAL.JAVA AND TREERANDOMNODE.JAVA ! They were missing methods and made the build fail. Looks like that part isnt finished yet. Just added standard methods to solve build errors !

File: src/main/java/com/thealgorithms/datastructures/trees/TreeTraversal.java [hunk 1]
```diff
@@ -104,16 +106,16 @@ public void printPostOrder() {
      */
     public void printLevelOrder() {
         LinkedList<Node> queue = new LinkedList<>();
-        queue.add(this);
+        queue.add(null); // What node should be added here ? This had a wrong parameter type
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/trees/TreeTraversal.java:109]: NodeTree

File: src/main/java/com/thealgorithms/datastructures/trees/TreeRandomNode.java [hunk 2]
```diff
@@ -17,33 +16,62 @@ the inOrder() method to store the values in the arraylist, then find the size of
...
-    Node left, right;
+    NodeTree left, right;
+	public String data;
 
     public Node(int key) {
         item = key;
         left = right = null;
     }
+
+	public void insert(int value) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/trees/TreeRandomNode.java:36]: Why have you added those methods? They are empty and, thus, useless

File: src/main/java/com/thealgorithms/datastructures/lists/CircleLinkedList.java [hunk 3]
```diff
@@ -63,8 +104,12 @@ public String toString() {
         return s + " ]";
     }
 
+    /** main(String args[]): <br>
```
[the remark this answers is not in the document]
Comment (Speecker) [reply src/main/java/com/thealgorithms/datastructures/lists/CircleLinkedList.java:107]: the javadoc for main() or the method itself ? there wont be a way to test the class without main if you dont call the classes methods elsewhere in another main(). Btw it was already there. wasnt implemented by me.

File: src/main/java/com/thealgorithms/datastructures/trees/TreeRandomNode.java (hunk 2 above)
Comment (Speecker) [reply src/main/java/com/thealgorithms/datastructures/trees/TreeRandomNode.java:36]: yes as i said someone wrote the calls to them but those called methods themselves werent declared anywhere. i just implemented them blanc just to have the full project compile without errors. i guess its up to the one who wrote the calls to finish their methods as he should know how he meant to implement them.

File: src/main/java/com/thealgorithms/datastructures/trees/TreeTraversal.java (hunk 1 above)
Comment (Speecker) [reply src/main/java/com/thealgorithms/datastructures/trees/TreeTraversal.java:109]: there is a left and a right node and if you want to use the NodeTree using this u still have to tell what part of the object to use as given parameter. The method add() asks for type Node e and not an Object. thats how far i understand it.
Either you may change the methods input parameter to Object or use a Node e. NodeTree would be an object.
I can help where i can but im still learning the stuff at my university :)

File: src/main/java/com/thealgorithms/datastructures/lists/CircleLinkedList.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (siriak) [reply src/main/java/com/thealgorithms/datastructures/lists/CircleLinkedList.java:107]: I see, I'm talking about the whole main(). It's a library, not an application, so it doesn't need an entry point. You can test it by automated tests, that's the way.

File: src/main/java/com/thealgorithms/datastructures/trees/TreeTraversal.java (hunk 1 above)
Comment (siriak) [reply src/main/java/com/thealgorithms/datastructures/trees/TreeTraversal.java:109]: You can make the queue of type LinkedList<TreeNode>

File: src/main/java/com/thealgorithms/datastructures/trees/TreeRandomNode.java (hunk 2 above)
Comment (siriak) [reply src/main/java/com/thealgorithms/datastructures/trees/TreeRandomNode.java:36]: Where are they called?

**Pull request closed without merging**